packages/react-lowcode/src/codegen/index.ts

```diff
@@ -47,8 +47,26 @@ export function generatePages(inputSourceCode: string, io: CodeRW & CodeDir, opt
                 true,
                 ts.ScriptKind.TSX
             )
-            const pageSouceCode = printer.printList(ts.ListFormat.MultiLine, factory.createNodeArray([...page.imports, page.functionDeclaration]), sourceFile)
+            const pageSouceCode = printer.printList(ts.ListFormat.MultiLine, factory.createNodeArray([...page!.imports, page!.functionDeclaration]), sourceFile)
             io.writeFile(filePath, pageSouceCode)
+
+            //generate list wrapper
+            const indexWrapperTemplatePath = 'path-to-template'//TODO: put here real template path when template will be done
+            let template = ''
+            io.readFile(indexWrapperTemplatePath).then((source => {if(source) template = source;}))
+
+            const listWrapper = generator.generateListComponentWrapper(template)
+            const listWrapperFilePath = `src/components/${typeName}Wrapper.tsx`//TODO: dont like the word wrapper, rename later to something else
+            const sourceFileWrapperSourceFile = ts.createSourceFile(
+                listWrapperFilePath,
+                '',
+                ts.ScriptTarget.ESNext,
+                true,
+                ts.ScriptKind.TSX
+            )
+
+            const wrapperPageSourceCode = printer.printList(ts.ListFormat.MultiLine, factory.createNodeArray([...listWrapper!.imports, listWrapper!.functionDeclaration]), sourceFileWrapperSourceFile)
```

Comment:
Why `generateListComponentWrapper` or `generateListPage` returns undefined? You should not write `!.`, rather then that write `?.` would be better in most cases.